aixplain/processes/data_onboarding/process_media_files.py

```diff
@@ -0,0 +1,223 @@
+__author__ = "thiagocastroferreira"
+
+import logging
+import os
+import pandas as pd
+import shutil
+import tarfile
+
+from aixplain.enums.data_type import DataType
+from aixplain.enums.file_type import FileType
+from aixplain.enums.storage_type import StorageType
+from aixplain.modules.file import File
+from aixplain.modules.metadata import MetaData
+from aixplain.utils.file_utils import upload_data
+from pathlib import Path
+from tqdm import tqdm
+from typing import List, Tuple
+
+
+def compress_folder(folder_path: str):
+    with tarfile.open(folder_path + ".tgz", "w:gz") as tar:
+        for name in os.listdir(folder_path):
+            tar.add(os.path.join(folder_path, name))
+    return folder_path + ".tgz"
+
+
+def run(metadata: MetaData, paths: List, folder: Path, batch_size: int = 100) -> Tuple[List[File], int, int, int, int]:
+    """Process a list of local media files, compress and upload them to pre-signed URLs in S3
+
+    Explanation:
+        Each media on "paths" is processed. If the media is in a public link, this link is added into an index CSV file.
+        If the media is in a local path, it will be copied into a local folder and its path will be added to the index CSV file.
+        The medias are processed in batches such that at each "batch_size" medias, the index CSV file is uploaded into a pre-signed URL in s3 and reset.
+        If the medias are stored locally, the local folder is compressed into a .tgz file and also uploaded into S3.
+
+    Args:
+        metadata (MetaData): meta data of the asset
+        paths (List): list of paths to local files
+        folder (Path): local folder to save compressed files before upload them to s3.
+
+    Returns:
+        Tuple[List[File], int, int, int]: list of s3 links; data, start and end columns index, and number of rows
+    """
+    # if files are stored locally, create a folder to store it
+    media_folder = Path(".")
```

Comment:
Usually, a good practice is to use a [temporary folder](https://docs.python.org/3/library/tempfile.html).
You can use a with context to create this folder and store the files in the tempdir. I would recommend that you delete this after uploading its contents.